spec/lib/spree/shipment_notice_spec.rb

```diff
@@ -45,6 +46,49 @@
           expect(notice.error).to be_present
         end
       end
+
+      context 'order is not paid' do
+        before do
+          expect(order).to receive(:paid?).and_return(false)
+        end
+
+        context 'capture at notification is not active' do
+          before do
+            Spree::Config.shipstation_capture_at_notification = false
+          end
+
+          it 'payments are not captured' do
+            expect(order).to_not receive(:capture_payments!)
+            expect(order).to receive(:errors)
+            expect(notice.apply).to eq(false)
+          end
+        end
+
+        context 'capture at notification is active' do
+          before do
+            Spree::Config.shipstation_capture_at_notification = true
+          end
+
+          it 'payments are captured' do
+            expect(notice).to receive(:update).and_return(true)
+            expect(order).to receive(:capture_payments!).and_return(true)
+            expect(notice.apply).to eq(true)
+          end
+
+          context 'capture fails' do
+            before do
+              expect(Rails.logger).to receive(:error)
+            end
+
+            it '#apply returns false and sets @error' do
+              expect(order).to receive(:capture_payments!).and_return(false)
+              expect(order).to receive(:errors)
+              expect(notice.apply).to eq(false)
+              expect(notice.error).to be_present
+            end
+          end
+        end
+      end
```

Comment:
This block of tests looks good. 

This `expect(Rails.logger).to receive(:error)` is a bit weird in the before block. It should either be an `allow`, or it should be in the `it` block.